vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+
+Now when you run the app, you should be seeing this -
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_1.png")
+```
+
+Note: If you check the built-in testcases once changes are introduced, running `rhino::test_r()` will give you errors.
+To avoid failed github actions for now (in case you are pushing your code to Github), you may empty the test cases present in `tests\testthat\test-main.R`.
+
+## 2. Show Data in a Table via a `rhino` module
+
+So, in this section, we will introduce a table to our application that displays random imaginary sales deals.
+The dataset comes from the `shiny.fluent` package itself - [fluentSalesDeals](https://appsilon.github.io/shiny.fluent/reference/fluentSalesDeals.html).
+
+### Creating the module
+
+-   In the `app\view\` folder, let us create a new file and name it `datatable.R`
+
+-   This `app\view\datatable.R` file will be used to display the data table on the UI and we will call this module in the `app/main.R` file once done.
+    Here is the code
+
+    ```         
+    box::use(
+      shiny[moduleServer, NS],
+      shiny.fluent[fluentPage, Text, DetailsList]
+    )
+
+    #' @export
+    ui <- function(id) {
+      ns <- NS(id)
+        fluentPage(
+            Text("A randomly generated dataset of imaginary sales deals"),
+            DetailsList(items = shiny.fluent::fluentSalesDeals)
+        )
+    }
+
+    #' @export
+    server <- function(id) {
+      moduleServer(id, function(input, output, session) {
+      })
+    }
+    ```
+
+So, here we are using [`DetailsList`](https://appsilon.github.io/shiny.fluent/reference/DetailsList.html) function to render the table.
+
+### Calling the module
+
+-   Go to `app/main.R` file and edit it with the following code
+
+```         
+box::use(
+  shiny[moduleServer, NS],
+  shiny.fluent[fluentPage, Text]
+)
+
+box::use(
+  app/view/datatable,
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  fluentPage(
+    Text("Hello!", variant = "mega"),
+    datatable$ui(ns("datatable"))
+  )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    datatable$server("datatable")
+  }
+  )
+}
+```
+
+Notice here that we have called the `datatable` module by importing it to our main application file first.
+It is now, that the main module will be able to use exported functions from `app/view/datatable.R`.
+
+This code in the `app/main.R` file uses the concept of Shiny modules.
+In the **`ui`** function, **`ns`** is used to create a unique namespace for the **`datatable`** element using the **`ns("datatable")`** syntax.
+This is important because when the UI contains multiple instances of the same element (e.g. multiple tables), each instance must have a unique namespace to prevent naming conflicts in the server function.
```

Comment:
Instead of this section, just mention that Rhino applications use Shiny modules and refer to the documentation in Rhino and Shiny.

Reply:
* Added reference to documentations
* Keeping the `ns` explanation here, removed from other parts of the tutorial